server/gui/main.py

```diff
@@ -237,13 +286,208 @@ def onError(self, err, server_error=False):
+        self.form_layout = QFormLayout()
+        self.form_widget = QFrame()
+        hidden = QSizePolicy()
+        hidden.retainSizeWhenHidden()
+        self.form_widget.setSizePolicy(hidden)
+        self.setupHeader()
+        self.setupForm()
+        self.setLayout(self.options_layout)
+
+    def reset(self):
+        self.title_widget.setText("")
+        self.embedding_widget.setCurrentIndex(0)
+        self.obs_widget.setText("")
+        self.var_widget.setText("")
+        self.max_category_items_widget.setValue(100)
+        self.diff_exp_lfc_cutoff_widget.setValue(0.1)
+        self.form_visibility_toggle.setIcon(self.icons["collapsed"])
+        self.form_widget.hide()
+
+    def setupHeader(self):
+        self.header_layout.addWidget(QLabel("Advanced Options"))
+        self.form_visibility_toggle = QPushButton()
+        self.form_visibility_toggle.setIcon(self.icons["expanded"])
+        self.form_visibility_toggle.clicked.connect(self.toggleOptionsVisibilitiy)
+        self.header_layout.addWidget(self.form_visibility_toggle)
+        self.header_layout.setStretch(0, 4)
+        self.options_layout.addLayout(self.header_layout)
+
+    def setupForm(self):
+        # Option Widgets
+        self.title_widget = QLineEdit()
+        self.title_widget.setAccessibleName("title")
+        self.embedding_widget = QComboBox(self)
+        self.embedding_widget.addItems([""] + MODES)
+        self.embedding_widget.setAccessibleName("layout")
+        self.obs_widget = QLineEdit()
+        self.obs_widget.setAccessibleName("obs_names")
+        self.var_widget = QLineEdit()
+        self.var_widget.setAccessibleName("var_names")
+        self.max_category_items_widget = QSpinBox()
+        self.max_category_items_widget.setAccessibleName("max_category_items")
+        self.max_category_items_widget.setRange(1, 100000)
+        self.max_category_items_widget.setValue(100)
+        self.diff_exp_lfc_cutoff_widget = QDoubleSpinBox()
+        self.diff_exp_lfc_cutoff_widget.setAccessibleName("diffexp_lfc_cutoff")
+        self.diff_exp_lfc_cutoff_widget.setRange(0, 1)
+        self.diff_exp_lfc_cutoff_widget.setValue(0.1)
+        self.diff_exp_lfc_cutoff_widget.setDecimals(3)
+        self.diff_exp_lfc_cutoff_widget.setStepType(QAbstractSpinBox.AdaptiveDecimalStepType)
+
+        # Connect change
+        self.title_widget.textChanged.connect(self.updateOpt)
+        self.embedding_widget.currentIndexChanged.connect(self.updateOpt)
+        self.obs_widget.textChanged.connect(self.updateOpt)
+        self.var_widget.textChanged.connect(self.updateOpt)
+        self.max_category_items_widget.valueChanged.connect(self.updateOpt)
+        self.diff_exp_lfc_cutoff_widget.valueChanged.connect(self.updateOpt)
+
+        # Labels and tooltips
+        title_label = QLabel("title:")
+        title_label.setToolTip("Title to display (if omitted will use file name)")
+        embedding_label = QLabel("embedding:")
```

Comment:
We should be consistent and use either embedding or layout, but not both.  Suggest using the same as the CLI option (currently `layout`)

Reply:
will fix